src/mesh/position.cxx

```diff
@@ -0,0 +1,97 @@
+#include "bout/sys/position.hxx"
+#include "boundary_region.hxx"
+#include "bout/constants.hxx"
+#include "bout/mesh.hxx"
+
+Position::Position(const BoundaryRegion* bndry, int iz, CELL_LOC loc, BoutReal _t, Mesh * msh)
+  : iz(iz), _t(_t), msh(msh), sx(loc == CELL_XLOW), sy(loc == CELL_YLOW), sz(loc == CELL_ZLOW) {
+  ix = bndry->x;
+  iy = bndry->y;
+  if (bndry->bx) {
+    sx = true;
+    if (bndry->bx < 0) {
+      ix += 1;
+    }
+  }
+  if (bndry->by) {
+    // iy is already outside
+    // Direction of the boundary [x+bx][y+by] is going outwards
+    // by < 0 -> same direction as staggering
+    // staggering info in boundary condition needs to be ignored
+    sy = true;
+    if (bndry->by < 0) {
+      iy += 1;
+    }
+  }
+};
+
+BoutReal Position::x() {
+  BoutReal fac = 1;
```

Comment:
Maybe consider:

```cpp
// Assuming the enum in Position becomes an enum class
const BoutReal normalisation{fx == Flag::TWOPI ? TWOPI : 1.};
```

Benefits:
1. `const` so the reader knows they only ever need to read this one line to know what the value is at any later point
2. descriptive name -- `fac` requires more thinking about than `normalisation`
3. removes the body of the `TWOPI` case